akka-actor-tests/src/test/java/akka/actor/JavaAPI.java

```diff
@@ -20,14 +20,13 @@
 
 import org.junit.ClassRule;
 import org.junit.Test;
-import org.scalatest.junit.JUnitSuite;
 import scala.Option;
 
 import java.util.Optional;
 
 import static org.junit.Assert.*;
 
-public class JavaAPI extends JUnitSuite {
+public class JavaAPI {
```

Comment:
I'm not sure if our tests which extend this class will still run with 'sbt test' after this change?

Reply:
if those tests are consistent about extending `JUnitSuiteLike`, I think it still works:

```
akka-actor-tests/src/test/scala/akka/actor/JavaAPISpec.scala:class JavaAPISpec extends JavaAPI with JUnitSuiteLike
akka-actor-tests/src/test/scala/akka/japi/JavaAPITest.scala:class JavaAPITest extends JavaAPITestBase with JUnitSuiteLike
```

and

```
akka > akka-actor-tests/testOnly *JavaAPI*
[info] JavaAPITest:
[info] - shouldBeSingleton
[info] - shouldCreateNone
[info] - shouldNotEnterForLoop
[info] - mustBeAbleToUseCurrentSystem
[info] - shouldEnterForLoop
[info] - shouldCreateSomeString
[info] - mustBeAbleToGetNoLogging
[info] JavaAPISpec:
[info] - mustBeAbleToCreateActorWIthBoxedAndUnBoxedConstructorParams
[info] - mustCompileTupleCreation
[info] - mustBeAbleToCreateActorRefFromFactory
[info] - mustBeAbleToCreateActorWIthNullConstructorParams2
[info] - mustBeAbleToCreateActorWIthNullConstructorParams3
[info] - mustBeAbleToCreateActorRefFromErasedFactory
[info] - mustBeAbleToCreateActorWIthConstructorParams
[info] - mustBeAbleToCreateActorRefFromClass
[info] - mustBeAbleToCreateOptionFromOptional
[info] - mustBeAbleToCreateActorWIthNullConstructorParams
[info] InboxJavaAPITest:
[info] - mustBeAbleToThrowTimeoutException
[info] ScalaTest
[info] Run completed in 2 seconds, 558 milliseconds.
[info] Total number of tests run: 18
[info] Suites: completed 3, aborted 0
[info] Tests: succeeded 18, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Passed: Total 18, Failed 0, Errors 0, Passed 18
[success] Total time: 6 s, completed Jun 17, 2019 1:02:43 PM
```

Reply:
This seems odd, but indeed mostly unrelated to this PR.

We've had some issues with running junit tests from sbt/scalatest over the years, sometimes they ran twice, sometimes not at all. Seems they were running twice before this PR, and with this change are running once - so that's good ;).

Even better would probably be to remove `JavaAPISpec` entirely and make `JavaAPI` extend `JUnitSuite` directly again - but that's for another PR